File: Data Structures & Algorithms/anagram-groups/submission-1.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        r = dict()
        for s in strs:
            c = [0]*26
            for i in s:
                c[ord(i)-ord('a')]+=1
            k = tuple(c)
            if k in r:
                r[k].append(s)
            else:
                r[k] = [s]
        return r.values()
        
        

File: Data Structures & Algorithms/string-encode-and-decode/submission-1.java
class Solution {
    private List<Integer> lenPrefixList;

    public Solution() {
        lenPrefixList = new ArrayList<>();
    }

    public String encode(List<String> strs) {
        for (String str : strs) {
            lenPrefixList.add(str.length());
        }

        return String.join("", strs);
    }

    public List<String> decode(String str) {
        List<String> strs = new ArrayList<>();
        int startIdx = 0;
        int endIdx = 0;

        for (Integer length : lenPrefixList) {
            startIdx = endIdx;
            endIdx = startIdx + length;

            strs.add(str.substring(startIdx, endIdx));
        }

        return strs;
    }
}
